outlines/caching.py

```diff
@@ -1,14 +1,17 @@
 import os
 from typing import Callable, Optional
-
+import hashlib
+import cloudpickle
+import torch
 from perscache import Cache, NoCache
```

Comment:
Will replace the perscache implementation, once you give a go-ahead on this one @rlouf.

Reply:
You mean in the OpenAI implementation?

Reply:
Yes.

Reply:
Oh ok, it doesn't work with async. Thanks for pointing that out.
ref: https://github.com/grantjenks/python-diskcache/issues/125

Reply:
It will work, the call to `diskcache` in the `async` function will be blocking but that's not an issue at all.

Reply:
Anyway, here is the method that you'd want to cache anyway: https://github.com/outlines-dev/outlines/blob/fa4be3aa1f15a6f10804fb8168e9223d7654b9b4/outlines/models/openai.py#L146

Reply:
My bad, it did work.